### PR TITLE
Add push-build support for OSX

### PR DESCRIPTION
**Pull request by shyamjvs**
Fix https://github.com/kubernetes/release/issues/656
Fix https://github.com/kubernetes/test-infra/issues/10459

/cc @BenTheElder @spiffxp @kubernetes/sig-release

### CODE REVIEW

File: lib/common.sh [hunk 1]
```diff
@@ -24,17 +24,19 @@ export PROG
...
-
 ##############################################################################
 # COMMON CONSTANTS
 #
-TOOL_LIB_PATH=${TOOL_LIB_PATH:-$(dirname $(readlink -ne $BASH_SOURCE))}
-TOOL_ROOT=${TOOL_ROOT:-$(readlink -ne $TOOL_LIB_PATH/..)}
+
+if [[ $(uname) == "Darwin" ]]; then
+  # Support for OSX.
+  TOOL_LIB_PATH=${TOOL_LIB_PATH:-$(dirname $(greadlink -f $BASH_SOURCE))}
```
Comment (ixdy) [inline lib/common.sh]: is `greadlink` GNU `readlink`? if so, shouldn't the args be the same for both OSX and non-OSX?

Comment (ixdy) [reply lib/common.sh]: right. but why are we doing `greadlink -f` here but `readlink -ne` below?

**Comment by shyamjvs** [comment]
@hoegaarden I can't comment about other release-related tooling, but this one definitely seems important for dev workflow for OSX devs who're using [kubetest](https://github.com/kubernetes/test-infra/issues/10459).

File: lib/common.sh (hunk 1 above)
Comment (shyamjvs) [reply lib/common.sh]: Oh, gotcha. Changed them to the same - PTAL.

**Comment by BenTheElder** [comment]
push-build is used for more than just official releases and this patch is quite small, I don't think we should actively shun macOS support! we could not focus on it, but we should accept PRs like this to fix it, imho.

File: lib/common.sh [hunk 2]
```diff
@@ -24,17 +24,22 @@ export PROG
 
 set -o errtrace
 
-# OSX not supported.  Tell them right away
+declare -A some_var || (echo "Running with unsupported bash version!" && exit 1)
```
Comment (BenTheElder) [inline lib/common.sh]: should we mention what version is supported?

Comment (ixdy) [reply lib/common.sh]: yeah, maybe say "bash 4.0 or greater required"?

Comment (shyamjvs) [reply lib/common.sh]: Done.

**Pull request merged**